Realm/RLMQueryUtil.mm

```diff
@@ -377,15 +427,18 @@ void add_between_constraint_to_query(realm::Query &query, std::vector<NSUInteger
             break;
         default:
             @throw RLMPredicateException(@"Unsupported predicate value type",
-                                         @"Object type %@ not supported for BETWEEN operations", RLMTypeToString(prop.type));
+                                         @"Object type %@ not supported for BETWEEN operations", RLMTypeToString(type));
     }
 }
 
 template<typename T>
 void add_binary_constraint_to_query(realm::Query & query,
                                     NSPredicateOperatorType operatorType,
-                                    NSUInteger index,
+                                    const ColumnReference& column,
                                     T value) {
+    RLMPrecondition(!column.has_links(), @"Unsupported operator", @"NSData properties cannot be queried over an object link.");
```

Comment:
Shouldn't this message read "Multi-level object equality link queries are not supported."?